src/select.rs

```diff
@@ -104,6 +104,10 @@ impl<T: Clone> PromptInteraction<T> for Select<T> {
     fn on(&mut self, event: &Event) -> State<T> {
         let Event::Key(key) = event;
 
+        if *key == Key::Escape {
+            return State::Cancel;
+        }
```

Comment:
...and it's because you changed the common interaction code... It will be reverted once you figure out how to modify the Input without changing the common interaction

Reply:
f587db5

Maybe the solution is not good, but it works at least.